### PR TITLE
Change code owners

### PR DESCRIPTION
**Pull request by owen-mc-diffblue**
This is based on top of https://github.com/diffblue/cbmc/pull/2580. This PR only concerns the last two commits. See the old PR for more discussion.

### CODE REVIEW

File: CODEOWNERS [hunk 1]
```diff
@@ -43,6 +43,5 @@
 /src/cpp/ @kroening @tautschnig @peterschrammel
 
 
-/scripts/ @diffblue/devops @thk123 @forejtv @peterschrammel
-/.travis.yml @diffblue/devops @thk123 @forejtv @peterschrammel
-/appveyor.yml @diffblue/devops @thk123 @forejtv @peterschrammel
+# All remaining files are assumed to be low-risk
+# Any two reviewers can be assigned
```
Comment (tautschnig) [inline CODEOWNERS]: To be honest, I wouldn't really consider any of the CI configuration files "low risk." Changes to them may have impact on the bills paid. Indeed the CodeBuild configuration files should be added here.  (I do agree that `/scripts/` is low risk.)

**Review by tautschnig — APPROVED** [review]
Works for me. I actually don't have much of an issue with the current status, so whatever others are happy with is ok for me.

**Comment by owen-mc-diffblue** [comment]
@diffblue/devops and @forejtv Are you okay with being removed as code owners from `-/scripts/`, `-/.travis.yml` and `-/appveyor.yml`?

**Review by allredj — COMMENTED** [review]
This PR failed Diffblue compatibility checks (cbmc commit: a81147ca2d36fc9ca5acf3ee363c197548bb49a6).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/79480656
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).
##

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: 52d71d2b97f85f67c24a041d71f7ce69e438c8c4).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/79482358

**Comment by smowton** [comment]
Personally I think we should leave the owner list as-is (i.e. the underlying #2580 should be merged, but drop this PR), but change the rules for how auto-review requests ought to be handled: When you receive an auto-review request, ASSIGN (cf. request-review-from) two people to do the actual reviewing (possibly yourself, possibly do it right now if it's trivial).

The problem behaviour is that people get auto-review requests and ignore by default. If your response to an auto-request is assign reviewers if not already assigned then the problem goes away.

**Comment by johnnonweiler** [comment]
As explained in #2580 this should not be merged in its current form.  Removing the default code owners has unintended consequences.

File: CODEOWNERS [hunk 2]
```diff
@@ -1,60 +1,47 @@
...
+/src/goto-cc/ @kroening @tautschnig
+/src/linking/ @kroening @tautschnig
+/src/memory-models/ @kroening @tautschnig
+/src/goto-symex/ @kroening @tautschnig @peterschrammel
+/src/json/ @kroening @tautschnig @peterschrammel
+/src/langapi/ @kroening @tautschnig @peterschrammel
+/src/xmllang/ @kroening @tautschnig @peterschrammel
+/src/nonstd/ @smowton @peterschrammel
+/src/solvers/cvc @martin-cs @kroening
+/src/solvers/flattening @martin-cs @kroening @tautschnig @peterschrammel
```
Comment (kroening) [inline CODEOWNERS:15]: /src/solvers/cvc is gone

File: CODEOWNERS [hunk 3]
```diff
@@ -23,24 +23,24 @@
...
-/src/util/ @smowton @kroening @tautschnig @peterschrammel @pkesseli
-/src/solvers/refinement @martin-cs @romainbrenguier @peterschrammel
-/jbmc/src/java_bytecode/ @smowton @mgudemann @thk123 @cristina-david @cesaro @pkesseli @NathanJPhillips @peterschrammel
-/src/analyses/ @martin-cs @peterschrammel @chrisr-diffblue @thk123 @smowton
-/src/pointer-analysis/ @martin-cs @peterschrammel @chrisr-diffblue @smowton
+/src/cbmc/ @kroening @tautschnig
+/src/goto-programs/ @kroening @tautschnig
+/src/util/ @kroening @tautschnig
+/src/solvers/refinement @romainbrenguier @peterschrammel
+/jbmc/src/java_bytecode/ @peterschrammel @smowton
```
Comment (thk123) [inline CODEOWNERS:30]: I think this isn't workable without someone from the Deeptest group as a code owner for `java_bytecode` as it needs to be changed frequently.

**Comment by owen-mc-diffblue** [comment]
There doesn't seem to be enough support for reducing the number of code owners for each folder. Closing this PR.

**Pull request closed without merging**